dvc/repo/experiments/queue/base.py

```diff
@@ -560,6 +561,7 @@ def _update_params(self, params: Dict[str, List[str]]):
         self.scm.add(list(params.keys()))
 
     @staticmethod
+    @retry(60, errors=LockError, timeout=1)
```

Comment:
The only question here is how long should we give for waiting on lock acquisition. It depends on both the number of workers and the time cost per initialization. But both of them are unknown.

Reply:
I think what we may actually want to do is revisit how we use `scm_locked`. Right now we lock the repo for the entire `init_executor` duration, but we actually only need to lock it for the first part of the method (getting the stash rev and dropping it).

Once we have finished modifying the stash rev we don't actually need to hold the lock any longer (`executor.init_git` and ``init_cache` do not need to lock the repo)

Reply:
adjusting the lock timeout and retry behavior is still needed, but it may also be better there to use celery task retries instead: https://docs.celeryq.dev/en/stable/userguide/tasks.html#automatic-retry-for-known-exceptions

Reply:
good catch, but we should make sure we are only locking the repo until we have pushed those refs (and not while we are pushing DVC cache).

Reply:
> good catch, but we should make sure we are only locking the repo until we have pushed those refs (and not while we are pushing DVC cache).

Yes but, we only set the local as cache dir for the remote python, no real date transfer happens in this progress.

Reply:
> no real date transfer happens in this progress.

For local experiments yes, if/when we get back to dvc machine that won't be the case